subprojects/core/src/main/java/org/gradle/internal/classpath/intercept/DefaultCallSiteDecorator.java

```diff
@@ -79,12 +79,7 @@ public DefaultCallSiteDecorator(List<CallInterceptor> callInterceptors) {
 
     private void addInterceptor(CallInterceptor interceptor) {
         for (InterceptScope scope : interceptor.getInterceptScopes()) {
-            CallInterceptor oldInterceptor = interceptors.put(scope, interceptor);
-            if (oldInterceptor != null) {
-                throw new IllegalArgumentException("Interceptor " + interceptor +
-                    " attempted to overwrite already registered " + oldInterceptor +
-                    " in the scope " + scope);
-            }
+            interceptors.compute(scope, (__, previous) -> previous == null ? interceptor : new CompositeCallInterceptor(previous, interceptor));
```

Comment:
💭 This is the main change: Before we failed if two interceptors intercepted the same scope (same method), now we compose interceptors via `CompositeCallInterceptor`.